cpp/include/cuspatial/experimental/detail/find/find_and_combine_segment.cuh

```diff
@@ -72,6 +73,8 @@ void find_and_combine_segment(OffsetRange offsets,
   auto [threads_per_block, num_blocks] = grid_1d(num_spaces);
   simple_find_and_combine_segments_kernel<<<num_blocks, threads_per_block, 0, stream.value()>>>(
     offsets, segments, merged_flag);
+
+  CUSPATIAL_CUDA_TRY(cudaGetLastError());
```

Comment:
This will not catch errors if the kernel has not thrown one yet. Synchronization is required. Therefore I think you should use this instead. In a release build, it's the same as what you have. But in a debug build, it synchronizes the stream.

```suggestion
  CUSPATIAL_CHECK_CUDA(stream.value());
```